slither/detectors/statements/unprotected_upgradeable.py

```diff
@@ -21,6 +21,23 @@ def _can_be_destroyed(contract) -> List[Function]:
     return targets
 
 
+def _has_initializer_modifier(functions: List[Function]) -> bool:
+    for f in functions:
+        for m in f.modifiers:
+            if m.name == "initializer":
+                return True
+    return False
+
+
+def _has_protected_initialize(functions: List[Function]) -> bool:
```

Comment:
Can you detail why adding this check? I think `_has_initializer_modifier` on the constructors should be sufficient to remove the FP's category

If there is a situation where `initializer` is not applied to the init function, I think this would be a separate access control issue, not related to the unprotected upgrade detector

Reply:
I added the check becasue I found odd that something like

```solidity
contract C is Initializable {
  constructor() initializer {}
  
  function initialize() external {
     require(owner == address(0));
     owner = msg.sender;
  }
  
  function kill() external {
   require(msg.sender == owner);
   // kill contract
  }
}
```
Would be skipped.

Reply:
I removed the `_has_protected_initialize` check.